src/message/attachment.rs

```diff
@@ -0,0 +1,137 @@
+use crate::message::{
+    header::{self, Header},
+    IntoBody, SinglePart,
+};
+use mime::Mime;
+
+#[derive(Clone, Copy)]
+enum Disposition {
+    Attachment,
+    Inline,
+}
+
+/// `SinglePart` builder for attachments
+///
+/// Allows building attachment parts easily.
+pub struct Attachment {
+    filename: Option<String>,
+    content_disposition: Disposition,
+    content_type: Option<Mime>,
+    content_id: Option<String>,
+}
+
+impl Default for Attachment {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Attachment {
+    /// Creates a new attachment
+    pub fn new() -> Self {
+        Self {
+            filename: None,
+            content_disposition: Disposition::Attachment,
+            content_type: None,
+            content_id: None,
+        }
+    }
+
+    /// Creates a new inline attachment
+    pub fn new_inline() -> Self {
+        Self {
+            filename: None,
+            content_disposition: Disposition::Inline,
+            content_type: None,
+            content_id: None,
+        }
+    }
```

Comment:
I'm not sure about this. With this being a higher level API it should probably offer better compile time guarantees. For example the `Content-Id` should always be present when the `Content-Disposition` is `inline`, same for `filename` for the attachment variant.
We may also want to require having a `Content-Type`, as I don't see a case where you would want it to be ambiguous anyway

Reply:
We could check more at compile time by restricting the features a bit while supporting most cases, with something like:

```rust
enum Disposition {
   Attachment(filename),
   Inline(content_id)
}

struct Attachment {
    body: IntoBody
    content_type: ContentType,
    disposition: Disposition
}

impl Attachment {
    new(filename) -> Self {}
    new_inline(content_id) -> Self {}
    body(body: IntoBody, content_type; ContentType) -> SinglePart
}
```
WDYT?

Reply:
I updated the branch with something like this